app/api/getAbsences/route.ts

```diff
@@ -0,0 +1,100 @@
+import { prisma } from '../../../utils/prisma';
+import { NextResponse } from 'next/server';
+
+export async function GET() {
+  try {
+    const absences = await prisma.absence.findMany({
+      select: {
+        id: true,
+        lessonDate: true,
+        subject: true,
+        lessonPlan: true,
+        reasonOfAbsence: true,
+        absentTeacher: {
+          select: {
+            id: true,
+            firstName: true,
+            lastName: true,
+            email: true,
+          },
+        },
+        substituteTeacher: {
+          select: {
+            id: true,
+            firstName: true,
+            lastName: true,
+            email: true,
+          },
+        },
+        location: {
+          select: {
+            id: true,
+            name: true,
+          },
+        },
+      },
+    });
+
+    if (!absences.length) {
+      return NextResponse.json({ events: [] }, { status: 200 });
+    }
+
+    const extractTimeArray = (date: Date) => [
+      date.getFullYear(),
+      date.getMonth() + 1,
+      date.getDate(),
+      date.getHours(),
+      date.getMinutes(),
+    ];
+
+    const events = absences.map((absence) => {
+      const attendees = [
+        {
+          name: `${absence.absentTeacher.firstName} ${absence.absentTeacher.lastName}`,
+          email: absence.absentTeacher.email,
+          rsvp: true,
+          partstat: 'TENTATIVE',
+          role: 'OPT-PARTICIPANT',
+        },
+      ];
+
+      if (absence?.substituteTeacher) {
+        attendees.push({
+          name: `${absence.substituteTeacher.firstName} ${absence.substituteTeacher.lastName}`,
+          email: absence.substituteTeacher.email,
+          rsvp: true,
+          partstat: 'TENTATIVE',
+          role: 'REQ-PARTICIPANT',
+        });
+      }
+
+      let lessonEndDate = absence.lessonDate;
+      lessonEndDate.setDate(lessonEndDate.getDate() + 1);
+
+      return {
+        start: extractTimeArray(absence.lessonDate),
+        end: extractTimeArray(lessonEndDate),
+        title: `${absence.absentTeacher.firstName} ${absence.absentTeacher.lastName}'s ${absence.subject}`,
```

Comment:
<img width="311" alt="image" src="https://github.com/user-attachments/assets/b4681d47-71c4-4dc7-a9c7-5caec40e6bfc">
 My event title as Object.Object, 
```suggestion
        title: `${absence.absentTeacher.firstName} ${absence.absentTeacher.lastName}'s ${absence.subject.name}`,
```